modules/gapi/test/internal/gapi_int_dynamic_graph.cpp

```diff
@@ -0,0 +1,215 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+#include "../test_precomp.hpp"
+
+#include <opencv2/gapi/cpu/core.hpp>
+#include <opencv2/gapi/cpu/imgproc.hpp>
+
+namespace opencv_test
+{
+    TEST(DynamicGraph, AddProtoInputArgs)
+    {
+        cv::GMat in1;
+        cv::GProtoInputArgs ins = GIn(in1);
+
+        cv::GMat in2;
+        EXPECT_NO_THROW(ins += GIn(in2));
+    }
+
+    TEST(DynamicGraph, AddProtoOutputArgs)
+    {
+        cv::GMat out1;
+        cv::GProtoOutputArgs outs = GOut(out1);
+
+        cv::GMat out2;
+        EXPECT_NO_THROW(outs += GOut(out2));
+    }
+
+    TEST(DynamicGraph, AddRunArgs)
+    {
+        cv::Mat out_mat1, out_mat2;
+        cv::GRunArgsP out_vector;
+        EXPECT_NO_THROW(out_vector += cv::gout(out_mat1));
+        EXPECT_NO_THROW(out_vector += cv::gout(out_mat2));
+    }
```

Comment:
You have checked output RunArgs, but why aren't you checking input RunArgs the same way? They differ slightly.
It's about this test and tests below where cv::gout() is checked, but gin() is not

Reply:
Yes they differ slightly.
Because of it += cv::gin() is not compiled for some reason
This feature is needed for the IFD sample, so I just left comment next to implementation to fix this

Reply:
+= cv::gin() is work now.